### ISSUE TITLE
Cobigen Templates and Update Mechanism need Versioning

### ISSUE THREAD

**Issue by Jan-PeterRichterCG** [devonfw/tools-cobigen#845]
### Expected behavior

As a developer that uses devon4j Version X.Y.Z, I want to download Cobigen Templates that match my devon4j version, which is not necessarily the latest one so that develoment / maintenance of older code is possible.
I want to do this using the Cobigen Eclipse-Plugin, i.e. I don't want to search for Cobigen Templates in the commit-history of a Git-Repository until I have found the commit that matches my devon4j release and then manually install templates (which I do not know how and where).

### Actual behavior
From eclipse (context menu from an Entity -> Cobigen -> Healt Check / Update Template),
- I can only check whether my Cobigen Templates are "current", i.e. the latest ones.
- I can only download the latest Cobigen Templates.

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. Try do download Cobigen Templates for devon4j 2.X.Y

**Comment by jdiazgon** [comment]
I Agree. One option can be:

* User clicks on `Update templates`.
* A dialog shows up with a `Download latest` button
* In case you don't want the latest ones, there will also be a textbox where you can specify the templates version to download.

Kind regards,

Jaime Díaz.

**Comment by Jan-PeterRichterCG** [comment]
Yes, but this requires that the Templates (or rather Template-sets) are managed in a versioned way on the Template repository side. I am afraid, that there is some work to do in order to identify the "most recend version of all Templates that correspond to devon4j Version X.Y.Z".
If the templates are managed within the same Git repo as all the rest of devon4j, this should not be so hard to do. But if not....

**Comment by maybeec** [comment]
The templates are published to maven central. So they are versioned and we even start with devon 3 to have version, which are compliant to the current release. The version scheme is 
<major-devon>.<minor-devon>.<bugfix-cobigen>

**Comment by Jan-PeterRichterCG** [comment]
Any action on this issue so far?

**Comment by maybeec** [comment]
@Jan-PeterRichterCG, as said, you could simply take the versioned templates from maven central: 
https://mvnrepository.com/artifact/com.devonfw.cobigen/templates-devon4j
https://mvnrepository.com/artifact/com.devonfw.cobigen/templates-oasp4j
https://mvnrepository.com/artifact/com.devonfw.cobigen/templates-oasp4js
